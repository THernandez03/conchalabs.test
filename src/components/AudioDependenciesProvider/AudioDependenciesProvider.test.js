import { describe, expect } from '@jest/globals';
import { createRef } from 'react';
import { render } from '@testing-library/react';

import {
  AudioDependenciesProvider,
  AudioDependenciesContext,
} from './AudioDependenciesProvider';

window.AudioContext = jest.fn(() => ({
  createMediaElementSource: () => 'mediaSource',
}));

const createComponent = ({ element }) => {
  const wrapper = ({ children }) => (
    <AudioDependenciesProvider audioElement={{ current: element }}>
      {children}
    </AudioDependenciesProvider>
  );
  const renderValues = jest.fn();

  render(
    <AudioDependenciesContext.Consumer>
      {(...args) => renderValues(...args)}
    </AudioDependenciesContext.Consumer>,
    { wrapper },
  );

  return renderValues;
};

describe('<AudioDependenciesProvider/>', () => {
  it('Should render the children', () => {
    const { container } = render(
      <AudioDependenciesProvider audioElement={createRef()}>
        Example
      </AudioDependenciesProvider>,
    );
    expect(container).toMatchSnapshot();
  });

  it('Should return empty values when no audioElement is passed as props', () => {
    const [firstRender] = createComponent({ element: null }).mock.calls;
    const [consumerValue] = firstRender;

    expect(consumerValue).toEqual({});
  });

  it('Should return valid value when audioElement is passed as props', () => {
    const [, secondRender] = createComponent({ element: <div /> }).mock.calls;
    const [consumerValue] = secondRender;

    expect(consumerValue).toMatchSnapshot();
  });
});
