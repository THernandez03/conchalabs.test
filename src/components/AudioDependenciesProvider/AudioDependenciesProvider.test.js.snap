// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<AudioDependenciesProvider/> Should render the children 1`] = `
<div>
  Example
</div>
`;

exports[`<AudioDependenciesProvider/> Should return valid value when audioElement is passed as props 1`] = `
Object {
  "audioContext": Object {
    "createMediaElementSource": [Function],
  },
  "audioElement": <div />,
  "mediaSource": "mediaSource",
}
`;
